src/graphql/schema/buildObjectType.ts

```diff
@@ -346,16 +349,18 @@ function buildObjectType({
                 id = relatedDoc.value;
               }
 
-              const result = await context.req.payloadDataLoader.load(JSON.stringify([
-                collectionSlug,
-                id,
-                0,
-                0,
-                locale,
-                fallbackLocale,
-                false,
-                false,
-              ]));
+              const result = await (context.req as PayloadRequest).payloadDataLoader.load(
+                {
+                  collection: collectionSlug as string,
```

Comment:
type is string | string[]

Can this be fixed?

Reply:
Yep - fixed here https://github.com/payloadcms/payload/tree/fix/dataloader-deadlock. Might have to pull this into your branch as I pushed, but my commit didn't show up in your original PR!